packages/openapi3/test/xml-models.test.ts

```diff
@@ -0,0 +1,686 @@
+import { deepStrictEqual } from "assert";
+import { describe, it } from "vitest";
+import { oapiForModel } from "./test-host.js";
+
+describe("@name", () => {
+  it("set the element value via @name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @name("xmlBook")
+        model Book {        
+          content: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        content: { type: "string" },
+      },
+      required: ["content"],
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+
+  it("set the element value for array property via @name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @name("xmlBook")
+        model Book {        
+          tags: string[];
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        tags: {
+          type: "array",
+          items: {
+            type: "string",
```

Comment:
i think this might need `xml.name: string` that's not how openapi would do it otherwise